admin/acceptance.py

```diff
@@ -562,6 +593,13 @@ def __init__(self):
             )
         self.creator = creator
 
+        if any(x not in string.ascii_letters + string.digits + '-'
+               for x in self.purpose):
+            raise UsageError(
+                "Purpose may have only alphanumeric symbols and dash. " +
+                "Found {!r}".format(self.purpose)
+            )
```

Comment:
I followed the example of validating `creator` here, but indeed `creator` is not a command-line parameter.